scripts/eqtl_hail_batch/launch_generate_eqtl_spearman.py

```diff
@@ -0,0 +1,150 @@
+        ).exists(cs_client)
+        return file_status
+
+    if cell_types is None:
+        # not provided (ie: use all cell types)
+        # we can infer the cell types from the 'expression_files'
+        # subdirectory of the input_path
+        # eg: {cell_type}_expression.tsv
+
+        path_to_expression_files = os.path.join(bucket_path, 'expression_files')
+        logging.info(f'Going to fetch cell types from {path_to_expression_files}')
+        blobs = bucket.list_blobs(prefix=path_to_expression_files + '/', delimiter='/')
+        ending = '_expression.tsv'
+
+        cell_types = [
+            os.path.basename(b.name)[:-len(ending)]
+            for b in blobs
+            if b.name.endswith('_expression.tsv')
+        ]
+        logging.info(f'Found {len(cell_types)} cell types: {cell_types}')
+
+    for cell_type in cell_types:
+        expression = os.path.join(
+            input_path, 'expression_files', f'{cell_type}_expression.tsv'
+        )
+        covariates = os.path.join(
+            input_path, 'covariates_files', f'{cell_type}_peer_factors.tsv'
+        )
+
+        if dry_run:
+            files_to_check = [expression, covariates]
+            files_that_are_missing = filter(
+                lambda x: not file_exists(x), files_to_check
+            )
+            for file in files_that_are_missing:
+                logging.error(f'File {file} is missing')
+
+        for chromosome in chromosomes:
+            geneloc = os.path.join(
+                input_path, 'gene_location_files', f'GRCh38_geneloc_chr{chromosome}.tsv'
+            )
+
+            if dry_run:
+                # check all files exist before running
+                files_to_check = [geneloc]
+                files_that_are_missing = filter(
+                    lambda x: not file_exists(x), files_to_check
+                )
+                for file in files_that_are_missing:
+                    logging.error(f'File {file} is missing')
+            else:
+
+                output_prefix = os.path.join(
+                    output_dir, f'{cell_type}', f'chr{chromosome}'
+                )
+                # add in sampleid keys
+                keys = os.path.join(input_path, 'OneK1K_CPG_IDs.tsv')
+                # The analysis-runner output path doesn't want the BUCKET specified,
+                # so let's remove it from the output_prefix
+                analysis_runner_output_path = output_prefix[5:].partition('/')[-1]
+                # get access level from bucket, rather than manual input
+                access_level = bucket_name.split('-')[-1]
+                run_analysis_runner(
+                    description=f'eqtl_spearman_{cell_type}_chr{chromosome}',
+                    dataset='tob-wgs',
+                    access_level=access_level,
+                    output_dir=analysis_runner_output_path,
+                    memory='highmem',
+                    cpu=8,
```

Comment:
Ditto with top comment, does this need 8cpus, or are you requiring a specific amount of memory?

Reply:
I'll go ahead and remove these